indico/modules/events/timetable/templates/display/_weeks.html

```diff
@@ -1,40 +1,164 @@
+                <li>
+                    <div class="tooltip-contrib">
+                        <span class="time">
+                            {{- contrib.start_dt|format_time(timezone=timezone) -}}
+                        </span>
+                        <span>
+                            <span class="title">{{ contrib.title }}</span>
+                            {% if contrib.speakers %}
+                                -
+                                <span>
+                                    {{ render_users(contrib.speakers, separator=' ', title=false) }}
+                                </span>
+                            {% endif %}
+                        </span>
+                    </div>
+                </li>
+            {% endfor %}
+        </ul>
+    </div>
 {% endmacro %}
 
 {% macro _render_tooltip(entry, timezone) %}
     <div class="tooltip hidden">
-        <strong>{{ entry.title }}</strong>
-        <br>
+        {% if entry.object.title %}
+            <span class="title">{{ entry.object.title }}</span>
+        {% endif %}
         {% set speakers = entry.object.person_links|selectattr("is_speaker")|list %}
+        {% if entry.type.name == 'CONTRIBUTION' and entry.object.track %}
+            <span class="track">
+                ({{ entry.object.track.title }})
+            </span>
+        {% endif %}
         {% if entry.type.name == 'CONTRIBUTION' and speakers %}
-             {{ render_users(speakers, span_class="compact-speakers", separator=' ', title=false) }}<br>
+            <ul>
+                {% for link in speakers -%}
+                    <li>{{ render_user_data(link, show_title=false, italic_affiliation=italic_affiliation) }}</li>
+                {% endfor %}
+            </ul>
         {% endif %}
-        {{ entry.start_dt | format_time(timezone=timezone) }} -
-        {{ entry.end_dt | format_time(timezone=timezone) }}
-        {% if entry.room_name %}
-            Room: {{ entry.room_name }}<br>
+        <div class="entry-time">
+            {{- entry.start_dt | format_time(timezone=timezone) }} -
+            {{ entry.end_dt | format_time(timezone=timezone) -}}
+        </div>
+        {% if entry.object.room_name %}
+            <span>Room: {{ entry.object.room_name }}</span>
         {% endif %}
-        {% if entry.object.session %}
-            Session: {{ entry.object.session.title }}<br>
+        {% if entry.type.name == 'CONTRIBUTION' %}
+            <span class="description"> {{ entry.object._description }}</span>
+        {% elif entry.type.name == 'SESSION_BLOCK' %}
+            {% if entry.object.contributions %}
+                {{ _render_contrib_list(entry.object) }}
+            {% endif %}
         {% endif %}
     </div>
 {% endmacro %}
 
+{% macro _render_time_slots(start_time, slot_entries, prev_end, hidden_count, short_contrib_count)%}
+    {% for entry, count in slot_entries %}
+        {% set __ = prev_end.append(entry.end_dt.astimezone(tz_object)) %}
```

Comment:
As I said the other day IRL, maybe we could move some of this stuff into macros, so that it's a little bit less confusing?